eng/packages/http-client-csharp/generator/Azure.Generator/test/Providers/TestData/ResourceProviderTests/Verify_ResourceProviderGeneration.cs

```diff
@@ -0,0 +1,117 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure;
+using Azure.Core;
+using Azure.Core.Pipeline;
+using Azure.ResourceManager;
+using Samples.Models;
+
+namespace Samples
+{
+    /// <summary></summary>
+    public partial class ResponseTypeResource : global::Azure.ResourceManager.ArmResource
```

Comment:
Lets trim this down to at most 1 method validation at a time.  Otherwise this test will be susceptible to any change in generation of any portion of a resource

Reply:
Is there a way to write a method of TypeProvider?
As I can see, there is no public API from TypeProviderWriter. Do we want to expose such API?

Reply:
The way we generate a specific method using mock provider depends on the override methods https://github.com/microsoft/typespec/blob/b742e1681b579a83e0887256c13dd66f688c4970/packages/http-client-csharp/generator/Microsoft.TypeSpec.Generator.ClientModel/test/TestHelpers/MockHelpers.cs#L94

Don't think it's worth to make AzureOutputLibrary.BuildResource `protected virtual` for testing. Will try to figure out another way of testing a single method.

Reply:
Separated the tests for ResourceClientProvider.
And after exposing AzureOutputLibrary.CreateResourceCore, I realized that we need to simplify the creation of ResourceClientProvider. And it ties to the contract of Input types from MGC.

Reply:
Isolated Resource related logic to ResourceBuilder.

Reply:
removed